vdom/core.py

```diff
@@ -63,12 +70,43 @@ class VDOM():
     >>> h1('Hey')
 
     """
+    _schema = VDOM_SCHEMA
 
-    def __init__(self, obj):
+    def __init__(self, obj, schema = None):
+        # we need to assign self.schema first,
+        # because it is used to validate the object
+        if schema is not None:
+            self.schema = schema
         self.obj = obj
 
     def _repr_mimebundle_(self, include, exclude, **kwargs):
-        return {'application/vdom.v1+json': toJSON(self.obj)}
+        return { 'application/vdom.v1+json': self.json_contents }
+
+    @property
+    def json_contents(self):
+        return to_json(self._obj, schema=self._schema)
+
+    @property
+    def obj(self):
+        return self._obj
+
+    @obj.setter
+    def obj(self, value):
+        if to_json(value, schema=self._schema) is not None:
+            self._obj = value
+
+    @property
+    def schema(self):
+        return self._schema
+
+    @schema.setter
+    def schema(self, value):
+        Draft4Validator.check_schema(value)
+        self._schema = value
+
+    @staticmethod
+    def validate(value, schema=_schema):
+        to_json(value, schema=schema)
```

Comment:
I was thinking the `VDOM` class was pointless prior to this (as we're able to use the raw elements for display). These helpers make it really nice though. Users could also subclass to make updating VDOM output (with updatable displays) really nice.

Reply:
Exactly what I was thinking; especially if people wanted to subclass it for more use-case specific schemas.

Reply:
Yeah, what really excited me here was the `obj` property.

Reply:
Sometimes I forget that my inline comments make it look like I'm only talking about the line above when sometimes I'm talking about a whole class. Next time I should put my review at the top of the class. 😄

Reply:
then I would just think that you're talking about my choice of what to inherit from 😄.

Also… you should always inherit from object rather than to use an empty pair of parens. That was the source of the py2/py3 bug (as crazy as that sounds).

Reply:
Ah that makes sense, I forgot that rule. My Python is rusty.

Reply:
No problem, I'd never run across an issue around importing from object or not before this bug, so I don't think it's terribly common to find yourself in a position where you're forced to attend to that.